assets/src/utils/dagre.ts

```diff
@@ -68,3 +71,36 @@ export function createDagre(
     },
   });
 }
+
+const graphInteractionEvents = ["click", "dragstart", "touchstart"];
+
+export function graphInteractionListener(
+  graph: Graph,
+  handler: () => void
+): () => void {
+  return function() {
+    graphInteractionEvents.forEach((event) => graph.on(event, () => {
+      handler();
+      graphInteractionEvents.forEach((event) => graph.off(event));
+    }));
+  };
+}
+
+export function nodeIdsDifferent(
+  nodes1: NodeConfig[],
```

Comment:
I would rename them to `previousNodes` `newNodes` or something like this, `nodes1` and `nodes2` looks really bad.

Reply:
Renamed in 3aa9e1c